sdk/schemaregistry/schema-registry-json/samples-dev/withEventHubsConsumerClient.ts

```diff
@@ -0,0 +1,130 @@
+import { JsonSerializer } from "@azure/schema-registry-json";
+import {
+  EventHubConsumerClient,
+  earliestEventPosition,
+  createEventDataAdapter,
+} from "@azure/event-hubs";
+
+// Load the .env file if it exists
+import * as dotenv from "dotenv";
+dotenv.config();
+
+// The fully qualified namespace for schema registry
+const schemaRegistryFullyQualifiedNamespace =
+  process.env["SCHEMA_REGISTRY_ENDPOINT"] || "<endpoint>";
+
+// The schema group to use for schema registeration or lookup
+const groupName = process.env["SCHEMA_REGISTRY_GROUP"] || "AzureSdkSampleGroup";
+
+// The connection string for Event Hubs
+const eventHubsConnectionString = process.env["EVENTHUB_CONNECTION_STRING"] || "";
+
+// The name of Event Hub the client will connect to
+const eventHubName = process.env["EVENTHUB_NAME"] || "";
+
+// The name of the Event Hub consumer group from which you want to process events
+const consumerGroup = process.env["CONSUMER_GROUP_NAME"] || "";
+
+// Sample Json Schema for user with first and last names
+const schemaObject = {
+  $schema: "https://json-schema.org/draft/2020-12/schema",
+  $id: "student",
+  type: "object",
+  properties: {
+    firstName: {
+      type: "string",
+    },
+    lastName: {
+      type: "string",
+    },
+  },
+  required: ["firstName", "lastName"],
+};
+
+const schema = JSON.stringify(schemaObject);
+
+// Description of the schema for registration
+const schemaDescription: SchemaDescription = {
+  name: `${schemaObject.$id}`,
+  groupName,
+  format: "Json",
+  definition: schema,
+};
+
+export async function main() {
+  // Create a new client
+  const schemaRegistryClient = new SchemaRegistryClient(
+    schemaRegistryFullyQualifiedNamespace,
+    new DefaultAzureCredential()
+  );
+
+  // Register the schema. This would generally have been done somewhere else.
+  // You can also skip this step and let `serialize` automatically register
+  // schemas using autoRegisterSchemas=true, but that is NOT recommended in production.
+  await schemaRegistryClient.registerSchema(schemaDescription);
+
+  // Create a new serializer backed by the client
+  const serializer = new JsonSerializer(schemaRegistryClient, {
+    groupName,
+    messageAdapter: createEventDataAdapter(),
+  });
+
+  const eventHubConsumerClient = new EventHubConsumerClient(
+    consumerGroup,
+    eventHubsConnectionString,
+    eventHubName
+  );
+
+  const subscription = eventHubConsumerClient.subscribe(
+    {
+      // The callback where you add your code to process incoming events
+      processEvents: async (events, context) => {
+        // Note: It is possible for `events` to be an empty array.
+        // This can happen if there were no new events to receive
+        // in the `maxWaitTimeInSeconds`, which is defaulted to
+        // 60 seconds.
+        // The `maxWaitTimeInSeconds` can be changed by setting
+        // it in the `options` passed to `subscribe()`.
+        for (const event of events) {
```

Comment:
Not needed here I think
```suggestion
        for (const event of events) {
```